Sources/SwiftGodot/Core/InspectableProperty.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  File.swift
+//  
+//
+//  Created by Marquis Kurt on 10/15/23.
+//
+
+import Foundation
+
+/// A structure that houses a property that can be added to a Godot inspector.
+public struct InspectableProperty<T> {
+    /// A typealias for the the method type used to register a property to Godot for inspectors.
+    public typealias RegisteredPropertyFunction = (T) -> ([Variant]) -> Variant?
+
+    /// The host object the property derives from.
+    public var hostObject: T.Type
+
+    /// The getter function that gets the property.
+    public var getter: RegisteredPropertyFunction
+
+    /// The setter function that sets the property.
+    public var setter: RegisteredPropertyFunction
```

Comment:
Could this and `getter` be `let` instead, then nothing here will be mutable.

Reply:
Sure, that can be arranged.

Reply:
`hostObject` too, if it's sticking around